rdbms/src/main/java/io/spine/server/storage/jdbc/table/entity/RecordTable.java

```diff
@@ -0,0 +1,234 @@
+        super(entityClass, StandardColumn.id.name(), dataSource);
+        this.typeRegistry = columnTypeRegistry;
+        queryFactory = new RecordStorageQueryFactory<>(dataSource,
+                                                       entityClass,
+                                                       log(),
+                                                       getIdColumn(),
+                                                       columnTypeRegistry);
+    }
+
+    @Override
+    public StandardColumn getIdColumnDeclaration() {
+        return StandardColumn.id;
+    }
+
+    @Override
+    protected ReadQueryFactory<I, EntityRecord> getReadQueryFactory() {
+        return queryFactory;
+    }
+
+    @Override
+    protected WriteQueryFactory<I, EntityRecordWithColumns> getWriteQueryFactory() {
+        return queryFactory;
+    }
+
+    @Override
+    protected List<TableColumn> getTableColumns() {
+        final List<TableColumn> columns = newLinkedList();
+        addAll(columns, StandardColumn.values());
+        final Collection<Column> entityColumns = EntityColumns.getColumns(getEntityClass());
+        final Collection<TableColumn> tableColumns = transform(entityColumns, new ColumnAdapter());
+        columns.addAll(tableColumns);
+        return columns;
+    }
+
+    public void write(Map<I, EntityRecordWithColumns> records) {
+        // Map's initial capacity is maximum, meaning no records exist in the storage yet
+
+        final Map<I, EntityRecordWithColumns> newRecords = new HashMap<>(records.size());
+
+        for (Map.Entry<I, EntityRecordWithColumns> unclassifiedRecord : records.entrySet()) {
+            final I id = unclassifiedRecord.getKey();
+            final EntityRecordWithColumns record = unclassifiedRecord.getValue();
+            if (containsRecord(id)) {
+                queryFactory.newUpdateQuery(id, unclassifiedRecord.getValue())
+                            .execute();
+            } else {
+                newRecords.put(id, record);
+            }
+        }
+        if (!newRecords.isEmpty()) {
+            queryFactory.newInsertEntityRecordsBulkQuery(newRecords).execute();
+        }
+    }
+
+    public Iterator<EntityRecord> readByQuery(EntityQuery<I> query, FieldMask fieldMask) {
+        final SelectByEntityColumnsQuery<I> queryByEntity =
+                queryFactory.newSelectByEntityQuery(query, fieldMask);
+        final Iterator<EntityRecord> result = queryByEntity.execute();
+        return result;
+    }
+
+    private final class ColumnAdapter implements Function<Column, TableColumn> {
+
+        private int columnNumber = 1;
+
+        @Override
+        public TableColumn apply(@Nullable Column column) {
+            checkNotNull(column);
+            final TableColumn result = new EntityColumnMapping(column, typeRegistry, columnNumber);
+            columnNumber++;
+            return result;
+        }
+    }
+
+    public enum StandardColumn implements TableColumn {
```

Comment:
I don't like `StandardColumn`. However, until it's documented, I am not able to suggest another name.